src/client/store/userManagement/actions/updateUserRoles.ts

```diff
@@ -0,0 +1,29 @@
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import { Functions } from '@utils/functions'
+import axios from 'axios'
+import { Dispatch } from 'redux'
+
+import { ApiEndPoint } from '@meta/api/endpoint'
+import { RoleName, UserRole } from '@meta/user'
+
+import { UserManagementActions } from '../slice'
+
+type Params = { roles: Array<Partial<UserRole<RoleName>>>; userId: number }
+
+const postUserRoles = Functions.debounce(async (params: Params, dispatch: Dispatch) => {
+  try {
+    const { data } = await axios.post(ApiEndPoint.User.roles(), params)
+
+    dispatch(UserManagementActions.setUserToEdit(data))
```

Comment:
dont you have already this user to edit in the store? this is not needed

Reply:
I had some asynchronous issue with the Debounce function, so I decided to use the same logic of putOriginalDataPoint

Reply:
I need the user roles from the response because they contain the role ids